src/resources/utils/debounce.ts

```diff
@@ -0,0 +1,12 @@
+// Delays calling of func by timeout ms
+export function debounce(func: (...args: any[]) => void, timeout = 300) {
```

Comment:
Why is this a function and not a const?

Reply:
Because the declaration of const is used with arrow functions, and this is not an arrow function

Reply:
I refrained from using an arrow function here since binding is so important and I didn't want to cross any wires

Reply:
Isn't it possible to bind with arrow function?

Reply:
Doesn't matter that much since this is a standalone reusable function, I'll resolve it